src/backend/distributed/commands/role.c

```diff
@@ -88,24 +106,110 @@ PostprocessAlterRoleStmt(Node *node, const char *queryString)
 }
 
 
+/*
+ * PreprocessAlterRoleSetStmt actually creates the plan we need to execute for alter
+ * role set statement.
+ */
+List *
+PreprocessAlterRoleSetStmt(Node *node, const char *queryString)
+{
+	if (!EnableAlterRolePropagation)
+	{
+		return NIL;
+	}
+
+	AlterRoleSetStmt *stmt = castNode(AlterRoleSetStmt, node);
+	ErrorIfUnsupportedAlterRoleSetStmt(stmt);
+
+	EnsureCoordinator();
+	QualifyTreeNode((Node *) stmt);
+	const char *sql = DeparseTreeNode((Node *) stmt);
+
+	List *commandList = list_make1((void *) sql);
```

Comment:
Similar to what we have for `PostprocessAlterRoleStmt` I have added checks in the beginning:
```c
	if (!EnableAlterRolePropagation || !IsCoordinator())
	{
		return NIL;
	}
```

Does this look good? Or should I use `DISABLE|ENABLE_DDL_PROPAGATION` instead?

I think we should change them both, for consistency. Maybe in a seperate PR, once this gets merged? What do you think @thanodnl ?

Reply:
ah, so it works because of the `!IsCoordinator()`, in other ddl commands we would error if it is not the coordinator I believe. IIRC we call `EnsureCoordinator()` to do this for us.

I think it is good to add the disable and enable ddl propagation commands in the `commandList` we are building here.
 a) this stays consistent between the propagations of ddl commands
 b) with `EnsureCoordinator` it gives good information to users that have ddl propagation on and try to run ddl commands on (mx) workers.